src/Abstractions/NexusMods.Abstractions.Loadouts/Mods/ModCategory.cs

```diff
@@ -0,0 +1,13 @@
+namespace NexusMods.Abstractions.Loadouts.Mods;
+
+public enum ModCategory
+{
+    GameFiles,
+    Mod,
+    Saves,
+    Overrides,
+    /// <summary>
+    /// Files maintained by the game's framework, not intended for direct user modification.
+    /// </summary>
+    Metadata,
+}
```

Comment:
Our previous iteration of categories had an open set, this is a closed set. We might want to discuss if this is the best approach. If we want to keep this as an enum, then all of these items should have explicit values.

@Al12rs thoughts?

Reply:
If this is a closed set we are only going to use this for GameFiles and Overrides and promoting those mods as even more special.  Games will not able to define their own special mods. So yeah I think this should be an open set.
I thought before we could have multiple categories on a mod?

Reply:
https://github.com/Nexus-Mods/NexusMods.App/issues/1315